### PR TITLE
Deprecate overwritepropertiesconverter, only used by non builder distributions

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/6294

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+### 🚩 Deprecations 🚩
+
+- Deprecate `overwritepropertiesconverter`, only used by non builder distributions. (#6295)
+- Deprecate `--set` flag, users can check the console for instructions on how to replace every call. (#6295)
+  - e.g. `-set=processors.batch.timeout=2s` => `-config=yaml:processors::batch::timeout: 2s`.
```
Comment (mx-psi) [inline CHANGELOG.md:9]:
```suggestion
  - e.g. `--set=processors.batch.timeout=2s` => `--config=yaml:processors::batch::timeout: 2s`.
```

File: service/flags.go [hunk 2]
```diff
@@ -47,13 +48,23 @@ func (s *stringArrayValue) String() string {
...
 			" has a higher precedence. Array config properties are overridden and maps are joined, note that only a single"+
-			" (first) array property can be set e.g. --set=processors.attributes.actions.key=some_key. Example --set=processors.batch.timeout=2s")
+			" (first) array property can be set e.g. --set=processors.attributes.actions.key=some_key. Example --set=processors.batch.timeout=2s", func(s string) error {
+			idx := strings.Index(s, "=")
+			if idx == -1 {
+				// No need for more context, see TestSetFlag/invalid_set.
+				return errors.New("missing equal sign")
+			}
+			cfgFlag := "yaml:" + strings.TrimSpace(strings.ReplaceAll(s[:idx], ".", "::")) + ": " + strings.TrimSpace(s[idx+1:])
+			_, _ = fmt.Fprint(flagSet.Output(), "-set flag is deprecated, replace `-set=", s, "` with `--config=", cfgFlag, "`\n")
```
Comment (mx-psi) [inline service/flags.go:65]:
```suggestion
			_, _ = fmt.Fprint(flagSet.Output(), "--set flag is deprecated, replace `--set=", s, "` with `--config=", cfgFlag, "`\n")
```

**Review by mx-psi — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bogdandrutu) [reply CHANGELOG.md]: Done.

File: service/flags.go (hunk 2 above)
Comment (bogdandrutu) [reply service/flags.go]: Done.

**Review by mx-psi — APPROVED** [review]
LGTM, looks like chloggen is broken? cc @codeboten

**Comment by bogdandrutu** [comment]
@mx-psi @codeboten found the error, the IDE validated the yaml and was green, but apparently the tool does not like

```
note: "Deprecate `--set` flag, users can check the console for instructions on how to replace every call, e.g. `--set=processors.batch.timeout=2s` => `--config=yaml:processors::batch::timeout: 2s`".
```

The last dot was outside the quotes, when moved inside worked.

**Comment by bogdandrutu** [comment]
@mx-psi still the tool uses a bad yaml parser, all the tools I found only are capable of reading that yaml, and say that it is valid. Anyway, not a huge issue.

File: .chloggen/deprecate-set-flag.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: service
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Deprecate `--set` flag, users can check the console for instructions on how to replace every call, e.g. `--set=processors.batch.timeout=2s` => `--config=yaml:processors::batch::timeout: 2s`."
```
Comment (dmitryax) [inline .chloggen/deprecate-set-flag.yaml:8]: The new interface doesn't seem very convenient to me. I know that it's good for promoting the same consistent configuration approach, but I would like to discuss it a bit more before merging this. Replied in the issue

Comment (bogdandrutu) [reply .chloggen/deprecate-set-flag.yaml]: This PR, does not touch "--set" flag anymore, so PTAL

**Review by codeboten — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**